### ISSUE TITLE
Reproducibility Notebook

### ISSUE THREAD

**Issue by lmazzon** [ageron/handson-ml#288]
hi,
I'm obtaining reproducible results for the Estimator API that are slightly different from yours:

INFO:tensorflow:Loss for final step: **0.13982485**.
<tensorflow.python.estimator.canned.dnn.DNNClassifier at 0x14de868fd68>

The final loss should be exactly 0.04650943.

INFO:tensorflow:Loss for final step: **1.1618254**.
<tensorflow.python.estimator.canned.dnn.DNNClassifier at 0x14df4e5bcf8>

The final loss should be exactly 1.0556093.

Conversly the Keras API is ok:
Epoch 10/10
60000/60000 [==============================] - 10s 166us/step - loss: 0.1001 - acc: 0.9716
<tensorflow.python.keras.callbacks.History at 0x14d9687a630>
You should get exactly 97.16% accuracy on the training set at the end of training.

My enviroment is:

>>> tf.__version__
 '1.10.0'
>>> np.__version__
 '1.14.5'

Python 3.6.2 |Continuum Analytics, Inc.| (default, Jul 20 2017, 12:30:02) [MSC v.1900 64 bit (AMD64)] on win32

Microsoft Windows [Versione 10.0.14393]

Thanks for your attention
Luigi

**Comment by lmazzon** [comment]
I obtain your same results:

>>> print(set("abcdefghijklmnopqrstuvwxyz0123456789"))
{'x', '2', 'i', 'r', 'p', 'd', '8', 'c', 'l', 'y', '3', 'h', 'm', 'z', '6', 'k', '5', 'o', 'a', '0', '7', 'g', '1', 'f', 'u', 'e', 'w', 'n', 'b', 'q', 'j', 't', 's', '4', '9', 'v'}

>>> import random; random.seed(42); print(random.random())
0.6394267984578837

>>> import numpy as np; np.random.seed(42); print(np.random.rand())
0.3745401188473625

>>> import tensorflow as tf; tf.enable_eager_execution(); tf.set_random_seed(42); print(tf.random_uniform(shape=[]))
tf.Tensor(0.6645621, shape=(), dtype=float32)

I can try a fresh python installation on Linux, which version of python, numpy & tensorflow would you suggest?

Thanks to you!
Luigi

PS thanks also for your excellent book and videos
